src/components/layout/Layout.tsx

```diff
@@ -3,18 +3,21 @@ import NavBar from "../navbar/NavBar";
 import $layout from "./Layout.module.css";
 import { useRouter } from "next/router";
 import { requiresLogin } from "@/client/lib/authenticated-page";
+import { useSession } from "next-auth/react";
 
 interface LayoutProps {
   children: React.ReactNode;
 }
 
 const Layout = ({ children }: LayoutProps) => {
   const router = useRouter();
+  const { status } = useSession();
   // TODO?: Are there better ways to achieve this?
   //Sets purple background for student profile and instructions routes
   const isPurpleBg =
     router.asPath.includes("/studentprofile") ||
     router.asPath.includes("/instructions");
+  const hasNavBar = "authenticated" == status;
```

Comment:
This replicates here the logic in the NavBar component to decide if the NavBar should be displayed:

https://github.com/sfbrigade/compass/blob/619dac8399708b8939479c3b126e9baae004a45f/src/components/navbar/NavBar.tsx#L109